src/main/java/duke/TaskHandler.java

```diff
@@ -0,0 +1,127 @@
+package duke;
+
+import exceptions.EmptyContentException;
+import exceptions.InvalidTaskAccessException;
+
+import java.util.List;
+
+public class TaskHandler {
+    private List<Task> content;
+    public TaskHandler(TaskList taskList) {
+        this.content = taskList.getContent();
+    }
+    private static final String HEADER = "Got it. I've added this task:";
+
+    public String display() {
+        String allElements = "";
+        for (int i = 0; i < content.size(); i++) {
+            if (! content.isEmpty()) {
+                allElements = allElements + (i + 1) + ". " + content.get(i).toString() + "\n";
+            } else {
+                break;
+            }
+        }
+        if (allElements == "") {
+            return "No items in list!";
+        } else {
+            return "Here are the tasks in your list:" + "\n" + allElements;
+        }
+    }
+
+    public String markAsDone(String input) {
+        int index = Parser.findIndex(input);
+        int listIndex = index + 1;
+        if (! content.isEmpty()) {
+            content.get(index).setDone();
+            return "Nice! I've marked this task as done: " + "\n" + listIndex + ". " + content.get(index).toString();
+        } else {
+            return "No such task.";
+        }
+    }
+
+    public String markAsUndone(String input) {
+        int index = Parser.findIndex(input);
+        int listIndex = index + 1;
+        if (content.isEmpty()) {
+            return "No such task.";
+        } else if (! content.isEmpty() && content.get(index).isDone) {
+            content.get(index).setUndone();
+            return "Nice! I've marked this task as undone: " + "\n" + (listIndex) + ". " + content.get(index).toString();
+        } else if(!content.get(index).isDone) {
+            return "This task is already marked undone. ";
+        } else {
+            return "No such task.";
+        }
+    }
+
+    public String eventHandler(String input) throws EmptyContentException {
```

Comment:
Notice that there are quite a few locations where method name does start with a verb. Perhaps consider doing so?